### ISSUE TITLE
Swig crashes during csharp bindings generation

### ISSUE THREAD

**Issue by grzelewski** [swig/swig#1897]
SWIG crashes when multiple enums defined in different namespaces have the same name.

example:

=== start crash.h ===

#ifndef DALI_ACTOR_ENUMERATIONS_H
#define DALI_ACTOR_ENUMERATIONS_H

namespace Dali {

namespace Dimension { enum Type { WIDTH  = 0x1, }; }
namespace ResizePolicy { enum Type { FIXED, }; }
namespace SizeScalePolicy { enum Type { USE_SIZE_SET, }; }

} // namespace Dali

#endif // DALI_ACTOR_ENUMERATIONS_H

=== END crash.h ===

SWIG config:

=== START crash.i ===

%module crash

%{
#include <crash.h>
%}

%include <crash.h>

=== END crash.i ===

How to run:

~/crash$ swig -c++ -csharp crash.i

Tar with examples:

[crash.tar.gz](https://github.com/swig/swig/files/5546079/crash.tar.gz)

**Comment by ojwb** [comment]
With SWIG 4.0.2 and git master at least, an error is reported first:

```
$ swig -c++ -csharp crash.i
crash.h:20: Error: 'Type' is multiply defined in the generated target language module.
:EOF: Error: Previous declaration of 'Type'
swig: ../../Source/Modules/csharp.cxx:1185: virtual int CSHARP::enumDeclaration(Node*): Assertion `!full_imclass_name' failed.
Aborted
```

The output could certainly be improved, but it does make it fairly clear what and where the issue is.

**Comment by wsfulton** [comment]
Fixed in 71b0073ff7892f40db09bb6bc2fa530bcf4be782
